go/vt/vttablet/tabletserver/semisyncwatcher/watcher.go

```diff
@@ -0,0 +1,282 @@
+/*
+Copyright 2025 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package semisyncwatcher
+
+import (
+	"context"
+	"errors"
+	"math"
+	"sync"
+	"time"
+
+	"vitess.io/vitess/go/timer"
+	"vitess.io/vitess/go/vt/dbconnpool"
+	"vitess.io/vitess/go/vt/log"
+	"vitess.io/vitess/go/vt/mysqlctl"
+	"vitess.io/vitess/go/vt/vttablet/tabletserver/tabletenv"
+)
+
+const (
+	semiSyncWaitSessionsRead = "SHOW STATUS LIKE 'Rpl_semi_sync_%_wait_sessions'"
+	semiSyncRecoverWrite     = "INSERT INTO semisync_recover (ts) VALUES (NOW())"
+	semiSyncRecoverClear     = "DELETE FROM semisync_recover"
```

Comment:
Any reason not to use truncate so that the table is dropped and re-created? It should be faster and then we have no bloat that builds up in the tablespace over time.

Reply:
Did not know I could do that 😆. TIL. Changed.